pkg/plugin/datasource.go

```diff
@@ -25,13 +29,41 @@ var (
 )
 
 // NewDatasource creates a new datasource instance.
-func NewDatasource(_ backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
-	return &Datasource{}, nil
+func NewDatasource(dataSourceSettings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+
+	log.DefaultLogger.Debug("settings from front end", "json data", dataSourceSettings.JSONData, "secure json data", dataSourceSettings.DecryptedSecureJSONData)
+	var settings map[string]string
+	err := json.Unmarshal(dataSourceSettings.JSONData, &settings)
+	if err != nil {
+		return nil, err
+	}
+	log.DefaultLogger.Debug("parsed json data", settings)
+	bucket := settings["bucket"]
+	id := dataSourceSettings.DecryptedSecureJSONData["aws_access_key_id"]
+	secretKey := dataSourceSettings.DecryptedSecureJSONData["aws_secret_access_key"]
+
+	cfg, err := config.LoadDefaultConfig(
+		context.TODO(),
```

Comment:
what is context.TODO()?  Is this something that's supposed to be changed out?

Reply:
it is a blank context instance. the context docs tell you not to pass a nil context if you don't have anything to configure in it so they provide context.TODO as  a default value. by passing this to a function it can add to your context without having to create it's own which would clobber any settings you wanted

https://pkg.go.dev/context